tests/cross/src/main/scala/tests/BasePCSuite.scala

```diff
@@ -1,105 +1,153 @@
-      .addDependencies(
-        Dependency(
-          mod"org.scala-lang:scala-library",
+  protected val index = new DelegatingGlobalSymbolIndex()
+
+  protected val workspace = new TestingWorkspaceSearch
+
+  private def indexScalaLibrary(
+      index: GlobalSymbolIndex,
+      scalaVersion: String
+  ): Unit = {
+    val sources = Fetch
+      .create()
+      .withClassifiers(Set("sources").asJava)
+      .withDependencies(
+        Dependency.of(
+          "org.scala-lang",
+          "scala-library",
           // NOTE(gabro): we should ideally just use BuildoInfoVersions.scalaVersion
           // but using the 2.11 stdlib would cause a lot tests to break for little benefit.
           // We can remove this switch once we drop support for 2.11
-          BuildInfoVersions.scalaVersion match {
+          scalaVersion match {
             case v if v.startsWith("2.13") => v
             case v if v.startsWith("2.12") => v
             case _ => BuildInfoVersions.scala212
           }
         )
       )
-      .run()
+      .fetch()
+      .asScala
     sources.foreach { jar => index.addSourceJar(AbsolutePath(jar)) }
   }
 
+  protected lazy val presentationCompiler: PresentationCompiler = {
+    val binaryVersion = ScalaDependencies.createBinaryVersion(scalaVersion)
+    val scalaDependencies =
+      if (binaryVersion.startsWith("0."))
+        ScalaDependencies.scala3(scalaVersion)
+      else
+        ScalaDependencies.scala2(scalaVersion)
+
+    val fetch = Fetch
+      .create()
+
+    val allDependencies = scalaDependencies ++ extraDependencies(scalaVersion)
+
+    allDependencies.foreach(fetch.addDependencies(_))
+    val libraries: Seq[Path] = fetch
+      .fetch()
+      .asScala
+      .map(_.toPath())
+
+    val myclasspath: Seq[Path] = libraries.filterNot { path =>
+      val filename = path.toString()
+      filename.contains("scala-reflect") ||
+      filename.contains("scala-compiler") ||
+      filename.contains("scala-xml")
+    }
+
+    if (requiresJdkSources)
+      JdkSources().foreach(jdk => index.addSourceJar(jdk))
+    if (requiresScalaLibrarySources)
+      indexScalaLibrary(index, scalaVersion)
+    val indexer = new Docstrings(index)
+    val search = new TestingSymbolSearch(
+      ClasspathSearch.fromClasspath(myclasspath),
+      new Docstrings(index),
+      workspace,
+      index
+    )
+
+    val scalacOpts = scalacOptions(myclasspath)
+
+    new ScalaPresentationCompiler()
+      .withSearch(search)
+      .withConfiguration(config)
+      .withExecutorService(executorService)
+      .withScheduledExecutorService(executorService)
+      .newInstance("", myclasspath.asJava, scalacOpts.asJava)
+  }
+
+  val tmp: AbsolutePath = AbsolutePath(Files.createTempDirectory("metals"))
+
+  override def afterAll(): Unit = {
+    presentationCompiler.shutdown()
+    RecursivelyDelete(tmp)
+    executorService.shutdown()
+  }
+
+  def testPc(name: TestOptions, ignoredScalaVersions: Set[String] = Set.empty)(
```

Comment:
I think we can replace `testPc` with the default MUnit `test` if we use test transforms. See following comments.

Do we need the `fn: PresentationCompiler => Unit` argument? It looks like we can access `presentationCompiler` directly

Also, what is the difference between an ignored scala version and excluded Scala version? My understanding is that `ignoredScalaVersions` is configurable per-test and `excludedScalaVersions` is configurable per-suite

Reply:
Removed it altogether, with the additional tags that you suggested it was no longer neccessary.